bee-common/bee-common/src/ord.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides features related to ordering.
+
+use core::{cmp::Ordering, iter::Iterator};
+
+/// Checks if an iterator yields ordered and unique values.
+pub fn is_unique_sorted<T: Ord, I: Iterator<Item = T>>(mut iterator: I) -> bool {
+    let mut previous = match iterator.next() {
+        Some(e) => e,
+        None => return true,
+    };
+
+    for curr in iterator {
+        if previous.cmp(&curr) != Ordering::Less {
```

Comment:
Why not `iter.is_sorted()` instead?

Reply:
It's nightly only

Reply:
Eurgh

Reply:
We'll remove when it's not anymore but in the meantime we kinda need this...